src/Native/Runtime/arm/StubDispatch.S

```diff
@@ -2,81 +2,165 @@
+RhpCastableObjectDispatch_CommonStubName:        .asciz "RhpCastableObjectDispatch_CommonStub"
+RhpTailCallTLSDispatchCellName:                  .asciz "RhpTailCallTLSDispatchCell"
+RhpCastableObjectDispatchHelper_TailCalledName:  .asciz "RhpCastableObjectDispatchHelper_TailCalled"
+RhpCastableObjectDispatchHelperName:             .asciz "RhpCastableObjectDispatchHelper"
+RhpVTableOffsetDispatchName:                     .asciz "RhpVTableOffsetDispatch"
+.text
+.macro GEN_ASSERT_FUNC func
+        GEN_ASSERT AssertMsg, FileName, \func
+.endm
+#endif
+
 
 LEAF_ENTRY RhpCastableObjectDispatch_CommonStub, _TEXT
-	// UNIXTODO: Implement this function
-	//int 3
+#ifdef _DEBUG
+        GEN_ASSERT_FUNC RhpCastableObjectDispatch_CommonStubName
+#else
+        // UNIXTODO: Implement this function
+        EMIT_BREAKPOINT
+#endif
 LEAF_END RhpCastableObjectDispatch_CommonStub, _TEXT
 
 LEAF_ENTRY RhpTailCallTLSDispatchCell, _TEXT
-	// UNIXTODO: Implement this function
-	// int 3
+#ifdef _DEBUG
+        GEN_ASSERT_FUNC RhpTailCallTLSDispatchCellName
+#else
+        // UNIXTODO: Implement this function
+        EMIT_BREAKPOINT
+#endif
 LEAF_END RhpTailCallTLSDispatchCell, _TEXT
 
 LEAF_ENTRY RhpCastableObjectDispatchHelper_TailCalled, _TEXT
-	// UNIXTODO: Implement this function
-	// int 3
+#ifdef _DEBUG
+        GEN_ASSERT_FUNC RhpCastableObjectDispatchHelper_TailCalledName
+#else
+        // UNIXTODO: Implement this function
+        EMIT_BREAKPOINT
+#endif
 LEAF_END RhpCastableObjectDispatchHelper_TailCalled, _TEXT
 
 LEAF_ENTRY RhpCastableObjectDispatchHelper, _TEXT
-	// UNIXTODO: Implement this function
-	// int 3
+#ifdef _DEBUG
+        GEN_ASSERT_FUNC RhpCastableObjectDispatchHelperName
+#else
+        // UNIXTODO: Implement this function
+        EMIT_BREAKPOINT
+#endif
 LEAF_END RhpCastableObjectDispatchHelper, _TEXT
 
+// Macro that generates a stub consuming a cache with the given number of entries.
+.macro DEFINE_INTERFACE_DISPATCH_STUB entries
+
+LEAF_ENTRY RhpInterfaceDispatch\entries, _TEXT
+
+          // r12 currently contains the indirection cell address. But we need more scratch registers and
+          // we may A/V on a null this. Both of these suggest we need a real prolog and epilog.
+          PROLOG_PUSH {r1-r2}
+
+          // load r2 to point to the cache block.
+          ldr     r2, [r12, #OFFSETOF__InterfaceDispatchCell__m_pCache]
+
+          // Load the EEType from the object instance in r0.
+          ldr     r1, [r0]
+
+          CurrentOffset = OFFSETOF__InterfaceDispatchCache__m_rgEntries
+          // For each entry in the cache, see if its EEType type matches the EEType in rax.
+          // If so, call the second cache entry.  If not, skip the InterfaceDispatchCacheEntry.
+          .rept \entries
+              ldr        r3, [r2, #CurrentOffset]
+              cmp        r1, r3
+              bne        0f
+              ldr        r3, [r2, #(CurrentOffset + 4)]
+              EPILOG_POP {r1-r2}
```

Comment:
Is the unwind info going to be correct for multiple `EPILOG_POP` like this? We may need to logically push `r1/r2` after the `bx r3` again, otherwise the unwinder will think that whatever follows `bx r3` is running with popped `r1/r2`.